internal/workflow.go

```diff
@@ -638,6 +638,7 @@ func (wc *workflowEnvironmentInterceptor) ExecuteActivity(ctx Context, typeName
 
 	if cancellable {
 		cancellationCallback.fn = func(v interface{}, more bool) bool {
+			assertNotInReadOnlyStateCancellation(ctx)
```

Comment:
I don't understand exactly. Are we using panics for control flow here? Is there any other way to not do stuff during coroutine exit?

Reply:
It's using panics for control flow as much as any of our other read only state checks or non determinism checks are.

Reply:
I assume what happens is we `recover()` in the panic handler but there is no workflow task to fail so it does nothing?

Is there any other/clearer way to not do stuff during coroutine exit?

Reply:
This assertion is almost the same as any of our other read only assertions we have in the Go SDK so it should be implemented as such.

Reply:
Is there any other/clearer way to not do stuff during coroutine exit? Can this be moved to where `cancellationCallback.fn` is invoked instead of put in each callback?

(just want to confirm that there is no better way than panicking at the beginning of every cancel callback)

Reply:
No, again this is the same thing as any of our other read only checks we added or non determinism. We always panic in those cases because it's the only option.